lua/astrocommunity/note-taking/global-note-nvim/init.lua

```diff
@@ -0,0 +1,49 @@
+local get_project_name = function()
+  local project_directory, err = vim.loop.cwd()
+  if project_directory == nil then
+    vim.notify(err, vim.log.levels.WARN)
+    return nil
+  end
+
+  local project_name = vim.fs.basename(project_directory)
+  if project_name == nil then
+    vim.notify("Unable to get the project name", vim.log.levels.WARN)
+    return nil
+  end
+
+  return project_name
+end
```

Comment:
```suggestion
local get_project_name = function()
  local project_name

  local project_directory, err = vim.loop.cwd()
  if not project_directory then
    vim.notify(err, vim.log.levels.WARN)
  else
    project_name = vim.fs.basename(project_directory)
  end

  if not project_name then
    vim.notify("Unable to get the project name", vim.log.levels.WARN)
  end

  return project_name
end
```

Recommendation to simplify logic of function for calculating the project name

Reply:
Yup, no problem with that ! FYI, the previous code was given by the author of the global-note.nvim repo :)